### DISCUSSION TITLE
lightweight webchat client available?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [processone/ejabberd#4294] [Questions]
I am looking for a lightweight webchat client for using with ejabberd. Not converse.js. That is 1MB js and has to many features. Just simple div and popout, like they show here[1]

[1]
On this page they are showing a webchat client 
https://www.process-one.net/en/solutions/instant_customer_support_chat/

**Comment by licaon-kter** [comment]
https://github.com/jsxc/jsxc ?

https://libervia.org/documentation/web ?

**Reply by f1-outsourcing** [reply]
> https://github.com/jsxc/jsxc ?

Looks like everything is from years ago, bosh is still being used while websockets are advised. If I run this example, I am getting some  JSXC is not defined.
From this screen captures it looks also not light, if I need to include everything that is in src, it is hundreds of files.

> https://libervia.org/documentation/web ?

looks like they use conversejs, I think if they had something of their own they would use that.

 I am looking just for a simple webchat window, no user lists no images or what ever.

**Reply by licaon-kter** [reply]
https://github.com/nioc/xmpp-web ?

**Reply by f1-outsourcing** [reply]
What is even vue? So I need to include yet another library? I just need to add something like this to an existing website. Can't this be done with 2-5 files?

![1](https://github.com/user-attachments/assets/af1f3bac-a8ae-4603-ada6-3f9224fa1708)
![2](https://github.com/user-attachments/assets/f96ed2f1-97d3-4407-8c67-0cf4e39c5c17)

**Reply by licaon-kter** [reply]
> Can't this be done with 2-5 files?

feel free to write one

please make if free software or open-source

drop a link here when done

thanks

**Reply by f1-outsourcing** [reply]
:) yes I am trying get it. I don't really get why nobody seems to have this. So much crap being added that is not needed for this.

**Reply by licaon-kter** [reply]
Famous last words
